### PR TITLE
Initial bzlmod support

### PR DESCRIPTION
**Pull request by jsharpe**
Initial changes required to support bzlmod.

### CODE REVIEW

File: toolchains/built_toolchains.bzl [hunk 1]
```diff
@@ -403,16 +404,17 @@ def _cmake_toolchain(version):
 
     fail("Unsupported cmake version: " + str(version))
 
-def _make_toolchain(version):
-    native.register_toolchains(
-        "@rules_foreign_cc//toolchains:built_make_toolchain",
-    )
+def _make_toolchain(version, register_toolchains):
+    if (register_toolchains):
```
Comment (abrisco) [inline toolchains/built_toolchains.bzl:408]: nit: Why is this wrapped in braces?

File: foreign_cc/repositories.bzl [hunk 2]
```diff
@@ -13,7 +13,8 @@ def rules_foreign_cc_dependencies(
         make_version = "4.3",
         ninja_version = "1.10.2",
         register_preinstalled_tools = True,
-        register_built_tools = True):
+        register_built_tools = True,
+        register_toolchains = True):
```
Comment (abrisco) [inline foreign_cc/repositories.bzl:17]: Why does registering the toolchains need to be optional for bzlmod support?

**Review by abrisco — APPROVED** [review]
A couple of questions but otherwise seems good to me 👍

File: toolchains/built_toolchains.bzl (hunk 1 above)
Comment (jsharpe) [reply toolchains/built_toolchains.bzl:408]: Context switching between too many languages!

File: foreign_cc/repositories.bzl (hunk 2 above)
Comment (jsharpe) [reply foreign_cc/repositories.bzl:17]: bzlmod removes the register_toolchains call and has its own mechanism for registering toolchains in the `MODULE.bazel` file. I haven't fully understood how we can make it configurable in the same way we currently do with `bzlmod`.

File: toolchains/built_toolchains.bzl [hunk 3]
```diff
@@ -424,10 +426,11 @@ def _make_toolchain(version):
 
     fail("Unsupported make version: " + str(version))
 
-def _ninja_toolchain(version):
-    native.register_toolchains(
-        "@rules_foreign_cc//toolchains:built_ninja_toolchain",
-    )
+def _ninja_toolchain(version, register_toolchains):
+    if (register_toolchains):
```
Comment (UebelAndre) [inline toolchains/built_toolchains.bzl:430]:
```suggestion
    if register_toolchains:
```

File: toolchains/built_toolchains.bzl (hunk 1 above)
Comment (UebelAndre) [reply toolchains/built_toolchains.bzl:408]:
```suggestion
    if register_toolchains:
```

File: toolchains/built_toolchains.bzl [hunk 4]
```diff
@@ -24,16 +24,17 @@ _CMAKE_SRCS = {
...
-def _cmake_toolchain(version):
-    native.register_toolchains(
-        "@rules_foreign_cc//toolchains:built_cmake_toolchain",
-    )
+    _cmake_toolchain(cmake_version, register_toolchains)
+    _make_toolchain(make_version, register_toolchains)
+    _ninja_toolchain(ninja_version, register_toolchains)
+
+def _cmake_toolchain(version, register_toolchains):
+    if (register_toolchains):
```
Comment (UebelAndre) [inline toolchains/built_toolchains.bzl:34]:
```suggestion
    if register_toolchains:
```

File: toolchains/prebuilt_toolchains.py [hunk 5]
```diff
@@ -141,9 +141,10 @@
 """
 
 REGISTER_TOOLCHAINS = """\
-native.register_toolchains(
+if (register_toolchains):
```
Comment (UebelAndre) [inline toolchains/prebuilt_toolchains.py:144]:
```suggestion
if register_toolchains:
```

**Review by UebelAndre — APPROVED** [review]
Looks good to me 👍

**Pull request merged**